crates/subspace-farmer/src/bin/subspace-farmer/commands/farm.rs

```diff
@@ -119,98 +120,80 @@ pub(crate) async fn farm_multi_disk(
         record_size.replace(farmer_protocol_info.record_size);
         recorded_history_segment_size.replace(farmer_protocol_info.recorded_history_segment_size);
 
-        let single_disk_farm = SingleDiskFarm::new(SingleDiskFarmOptions {
-            plot_directory: disk_farm.plot_directory,
-            metadata_directory: disk_farm.metadata_directory,
-            allocated_plotting_space: disk_farm.allocated_plotting_space,
-            farmer_protocol_info,
-            disk_concurrency,
-            archiving_client,
-            farming_client,
+        let single_disk_plot = SingleDiskPlot::new(SingleDiskPlotOptions {
+            directory: disk_farm.plot_directory,
+            allocated_space: disk_farm.allocated_plotting_space,
+            rpc_client: farming_client,
             reward_address,
-            bootstrap_nodes: bootstrap_nodes.clone(),
-            listen_on: listen_on.clone(),
```

Comment:
That and also because I think integration will be from the outside, node instance will likely be just given as an argument instead of being created internally and this PR is really not the place to figure that out.